---
title: VDJ
description: Single-cell VDJ analyses.
---


## Immunity overview

### Innate vs Adaptive immunity

![Just as resistance to disease can be innate (inborn) or acquired, the mechanisms mediating it can be correspondingly divided into innate (left) and adaptive (right), each composed of both cellular (lower half) and humoral elements (i.e. free in serum or body fluids; upper half). Adaptive mechanisms, more recently evolved, perform many of their functions by interacting with the older innate ones.](https://www.creative-diagnostics.com/upload/image/Innate%20and%20adaptive%20immune%20mechanisms.png)

![Innate and adaptive immunity time line. The mechanisms of innate immunity provide the initial defense against infections. Adaptive immune responses develop later and require the activation of lymphocytes. The kinetics of the innate and adaptive immune responses are approximations and may vary in different infections.](https://www.creative-diagnostics.com/upload/image/innate%20and%20adaptive%20immunity2.png)

[Source](https://www.creative-diagnostics.com/innate-and-adaptive-immunity.htm)

### Adaptive immunity

![The figure shows an adaptive immune response to the first and second encounter with the SARS-CoV-2 antigen (spike protein). Following the first encounter, various antigens go through the process of phagocytosis and decomposition inside the antigen-presenting cells (APCs). APCs fragment the antigen into smaller peptides, which they present on their surface mediated by surface receptors called major histocompatibility complex class II (MHC-II) molecules. The antigens are then presented to several types of cells in the host, among which we emphasize CD4+ T helper cells (also B and CD8+ cells). B cells which differentiate into plasma cells, secrete antibodies that inhibit the entry of the viral particle into the healthy cells. The activation of T helper cells by APC causes them to differentiate into different subtypes with specific functions mediated by cytokine secretion and cell-to-cell contact. Th2-differentiated T helper cells help humoral responses mature by providing a second signal to B cells, mostly through IL-4 secretion and CD40/CD40L interaction. Some CD4+ cells also become T follicular helper cells (Tfh), which govern the important interactions in the germinal centers important for the maturation of memory B cells and long-lived high-affinity antibody-producing plasma cells. Another subset of CD4+ T cells differentiates into a pool of memory T helper cells. Th1 T helper cells play a crucial role in cellular response formation. They pivot the MHC-I activation of CD8+ cells (CTL, cytotoxic T lymphocytes) by interacting with APC’s simultaneously. Activated CTLs then act by causing apoptosis (by Fasl ligand–FasL binding) of the host cells that are infected with the SARS-CoV-2. Some CTLs differentiate into memory cytotoxic T cells, which have the role of fast restoration of the CTL response with secondary antigen contacts. A similar mechanism of destruction occurs when NK cells interact with the virally infected cell. They contain granules with IFNγ and TNFα in their cytoplasm, which, when secreted, cause programmed cell death. As well, the mechanism of activation of NK cells does not occur through MHC molecules is important because MHC is not always present on the virally infected cells.](https://www.frontiersin.org/files/Articles/848582/fimmu-13-848582-HTML/image_m/fimmu-13-848582-g001.jpg)

@primorac2022adaptive

::: {#fig-cells-immune layout-ncol=2}

![](assets/kavathas-2019.png)

![](assets/kavathas-2019-2.png)

Cells of the immune system.

:::

@Kavathas2019, @LEWIS201919

### VDJ recombination

V(D)J recombination is the mechanism of somatic recombination that occurs only in developing lymphocytes during the early stages of T and B cell maturation. It results in the highly diverse repertoire of antibodies/immunoglobulins and T cell receptors (TCRs) found in B cells and T cells, respectively. The process is a defining feature of the adaptive immune system.

V(D)J recombination in mammals occurs in the primary lymphoid organs (bone marrow for B cells and thymus for T cells) and in a nearly random fashion rearranges variable (V), joining (J), and in some cases, diversity (D) gene segments. The process ultimately results in novel amino acid sequences in the antigen-binding regions of immunoglobulins and TCRs that allow for the recognition of antigens from nearly all pathogens including bacteria, viruses, parasites, and worms as well as "altered self cells" as seen in cancer.

### TCR review

- Human T cells develop in the thymus and they acquire the ability to recognize foreign antigens and provide protection against many different pathogens.
- This functional flexibility is guaranteed by the expression of highly polymorphic surface receptors called T cell receptors (TCRs).
- TCR is composed of two different protein chains. The vast majority of human T cells express TCRs composed of α (alpha) and β (beta) chains.
- The genes encoding alpha (TCRA) and beta (TCRB) chains are composed of multiple non-contiguous gene segments which include variable (V), diversity (D), and joining (J) segments for TCRB gene and variable (V) and joining (J) for TCRA gene.
- The enormous diversity of T cell repertoires is generated by random combinations of germ line gene segments (combinatorial diversity) and by random addition or deletion at the junction site of the segments that have been joined (junctional diversity).
- The sequence encoded by the V(D)J junction is called complementarity determining region 3 or CDR3. This sequence has the highest variability in both alpha and beta chains and determines the ability of a T cell to recognize an antigen.
- The total number of possible combination is estimated to exceed 10^18.
- The diversity of naïve T cells is the T cell repertoire
- Exposure to an antigen drives a rapid clonal expansion of cells carrying identical TCRs to generate a population of “effector cells.”
- After antigen clearance, a reduced number of these cells remain in the blood as “memory cells.”

![Somatic V(D)J arrangement in the alpha and beta chains. (A) Genomic organization and somatic recombination of TCRB and TCRA loci. Antigen repertoire diversity is guaranteed by a recombination step that progressively rearranges V, D, and J segments for T cell receptor (TCR) beta chains and V and J segments for TCR alpha chains. This variability (combinatorial diversity) is further increased by addition or deletion of nucleotides at the junction sites (junctional diversity). (B) Productive arrangements of beta and alpha transcripts. (C) Organization of TCR. TCR is composed by two subunits TCR alpha and TCR beta each organized in a constant region and a variable region responsible for antigen recognition.](https://www.frontiersin.org/files/Articles/384005/fimmu-09-01638-HTML/image_m/fimmu-09-01638-g001.jpg)

@de2018single

### B cell trajectories


@MORGAN2022210

[{{< fa brands youtube >}} Reconstruction and analysis of B cell lineage trees from single cell data using Immcantation ](https://www.youtube.com/watch?v=C0wnju8REvw)

## Tools

Tools related to VDJ analyses.

[[Collection of VDJ analyses tools](https://github.com/slowkow/awesome-vdj)]{.large}

**General**

- [{{< fa toolbox >}} Enclone (10X, Shell)](https://10xgenomics.github.io/enclone/) SHM-aware clonotyping, phylogenetic/lineage analysis, multiple sequence alignment
- [{{< fa toolbox >}} Immcantation (Shell)](https://immcantation.readthedocs.io/en/stable/) QC, UMI processing, clonal clustering, germline reconstruction, lineage topology, repertoire diversity, mutation profiling

**Reads to VDJ data**

- [{{< fa toolbox >}} CellRanger (Shell)](https://support.10xgenomics.com/single-cell-vdj/software/pipelines/latest/using/vdj)
- [{{< fa toolbox >}} Immcantation (Shell)](https://immcantation.readthedocs.io/en/stable/)
- [{{< fa toolbox >}} MIXCR (Shell)](https://github.com/milaboratory/mixcr) Start from any read data

**Other**

- [{{< fa toolbox >}} vdjtools (Java)](https://github.com/mikessh/vdjtools)
- [{{< fa toolbox >}} conga (Python)](https://github.com/phbradley/conga)
- [{{< fa toolbox >}} SONAR (Python)](https://github.com/scharch/SONAR)
- [{{< fa toolbox >}} scRepertoire (R)](https://github.com/ncborcherding/scRepertoire)
- [{{< fa toolbox >}} djvdj (R)](https://github.com/rnabioco/djvdj)

## References {.unnumbered}
