<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.340">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>RSKB - 6&nbsp; References</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<link href="../chapters/meta/index.html" rel="prev">
<link href="../assets/favicon.png" rel="icon" type="image/png">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>
<style>html{ scroll-behavior: smooth; }</style>
<link rel="preconnect" href="https://fonts.googleapis.com">
<link rel="preconnect" href="https://fonts.gstatic.com" crossorigin="">
<link href="https://fonts.googleapis.com/css2?family=JetBrains+Mono&amp;family=Nunito:ital,wght@0,400;0,600;1,400;1,600&amp;display=swap" rel="stylesheet">


</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../chapters/references.html"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">References</span></a></li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">References</span></h1>
                      </div>
  </div>
    
  
  <div class="quarto-title-meta">

      
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../">RSKB</a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/royfrancis/rskb" title="Source Code" class="quarto-navigation-tool px-1" aria-label="Source Code"><i class="bi bi-github"></i></a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Home</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/bulk/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Bulk RNA-Seq</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/sc/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Single-cell RNA-Seq</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/spatial/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Spatial RNA-Seq</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/meta/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Meta Transcriptomics</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/references.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">References</span></span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        
    </div>
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content">




<div id="refs" class="references csl-bib-body hanging-indent" data-line-spacing="2" role="list">
<div id="ref-argelaguet2021computational" class="csl-entry" role="listitem">
Argelaguet, R., Cuomo, A. S., Stegle, O., &amp; Marioni, J. C. (2021).
Computational principles and challenges in single-cell data integration.
<em>Nature Biotechnology</em>, <em>39</em>(10), 1202–1215.
</div>
<div id="ref-baran2018experimental" class="csl-entry" role="listitem">
Baran-Gale, J., Chandra, T., &amp; Kirschner, K. (2018). Experimental
design for single-cell RNA sequencing. <em>Briefings in Functional
Genomics</em>, <em>17</em>(4), 233–239.
</div>
<div id="ref-baruzzo2017simulation" class="csl-entry" role="listitem">
Baruzzo, G., Hayer, K. E., Kim, E. J., Di Camillo, B., FitzGerald, G.
A., &amp; Grant, G. R. (2017). Simulation-based comprehensive
benchmarking of RNA-seq aligners. <em>Nature Methods</em>,
<em>14</em>(2), 135–139.
</div>
<div id="ref-elixirbelgium2022" class="csl-entry" role="listitem">
Belgium, E. (2022). <em>Data volume</em>. <a href="https://rdm.elixir-belgium.org/data_volume.html">https://rdm.elixir-belgium.org/data_volume.html</a>
</div>
<div id="ref-cakir2020comparison" class="csl-entry" role="listitem">
Cakir, B., Prete, M., Huang, N., Van Dongen, S., Pir, P., &amp; Kiselev,
V. Y. (2020). Comparison of visualization tools for single-cell RNAseq
data. <em>NAR Genomics and Bioinformatics</em>, <em>2</em>(3), lqaa052.
</div>
<div id="ref-chhangawala2015impact" class="csl-entry" role="listitem">
Chhangawala, S., Rudy, G., Mason, C. E., &amp; Rosenfeld, J. A. (2015).
The impact of read length on quantification of differentially expressed
genes and splice junction detection. <em>Genome Biology</em>,
<em>16</em>(1), 1–10.
</div>
<div id="ref-conesa2016survey" class="csl-entry" role="listitem">
Conesa, A., Madrigal, P., Tarazona, S., Gomez-Cabrero, D., Cervera, A.,
McPherson, A., Szcześniak, M. W., Gaffney, D. J., Elo, L. L., Zhang, X.,
et al. (2016). A survey of best practices for RNA-seq data analysis.
<em>Genome Biology</em>, <em>17</em>(1), 1–19.
</div>
<div id="ref-corley2017differentially" class="csl-entry" role="listitem">
Corley, S. M., MacKenzie, K. L., Beverdam, A., Roddam, L. F., &amp;
Wilkins, M. R. (2017). Differentially expressed genes from RNA-seq and
functional enrichment results are affected by the choice of single-end
versus paired-end reads and stranded versus non-stranded protocols.
<em>BMC Genomics</em>, <em>18</em>(1), 1–13.
</div>
<div id="ref-de2018single" class="csl-entry" role="listitem">
De Simone, M., Rossetti, G., &amp; Pagani, M. (2018). Single cell t cell
receptor sequencing: Techniques and future challenges. <em>Frontiers in
Immunology</em>, <em>9</em>, 1638.
</div>
<div id="ref-dillies2013comprehensive" class="csl-entry" role="listitem">
Dillies, M.-A., Rau, A., Aubert, J., Hennequet-Antier, C., Jeanmougin,
M., Servant, N., Keime, C., Marot, G., Castel, D., Estelle, J., et al.
(2013). A comprehensive evaluation of normalization methods for illumina
high-throughput RNA sequencing data analysis. <em>Briefings in
Bioinformatics</em>, <em>14</em>(6), 671–683.
</div>
<div id="ref-evans2018selecting" class="csl-entry" role="listitem">
Evans, C., Hardin, J., &amp; Stoebel, D. M. (2018). Selecting
between-sample RNA-seq normalization methods from the perspective of
their assumptions. <em>Briefings in Bioinformatics</em>, <em>19</em>(5),
776–792.
</div>
<div id="ref-fu2018elimination" class="csl-entry" role="listitem">
Fu, Y., Wu, P.-H., Beane, T., Zamore, P. D., &amp; Weng, Z. (2018).
Elimination of PCR duplicates in RNA-seq and small RNA-seq using unique
molecular identifiers. <em>Bmc Genomics</em>, <em>19</em>, 1–14.
</div>
<div id="ref-fullgrabe2020guidelines" class="csl-entry" role="listitem">
Füllgrabe, A., George, N., Green, M., Nejad, P., Aronow, B., Fexova, S.
K., Fischer, C., Freeberg, M. A., Huerta, L., Morrison, N., et al.
(2020). Guidelines for reporting single-cell RNA-seq experiments.
<em>Nature Biotechnology</em>, <em>38</em>(12), 1384–1386.
</div>
<div id="ref-gallego2014rna" class="csl-entry" role="listitem">
Gallego Romero, I., Pai, A. A., Tung, J., &amp; Gilad, Y. (2014).
RNA-seq: Impact of RNA degradation on transcript quantification. <em>BMC
Biology</em>, <em>12</em>(1), 1–13.
</div>
<div id="ref-hart2013calculating" class="csl-entry" role="listitem">
Hart, S. N., Therneau, T. M., Zhang, Y., Poland, G. A., &amp; Kocher,
J.-P. (2013). Calculating sample size estimates for RNA sequencing data.
<em>Journal of Computational Biology</em>, <em>20</em>(12), 970–978.
</div>
<div id="ref-hsieh2019effect" class="csl-entry" role="listitem">
Hsieh, P.-H., Oyang, Y.-J., &amp; Chen, C.-Y. (2019). Effect of de novo
transcriptome assembly on transcript quantification. <em>Scientific
Reports</em>, <em>9</em>(1), 8304.
</div>
<div id="ref-kim2012short" class="csl-entry" role="listitem">
Kim, Y.-K., Yeo, J., Kim, B., Ha, M., &amp; Kim, V. N. (2012). Short
structured RNAs with low GC content are selectively lost during
extraction from a small number of cells. <em>Molecular Cell</em>,
<em>46</em>(6), 893–895.
</div>
<div id="ref-klepikova2017effect" class="csl-entry" role="listitem">
Klepikova, A. V., Kasianov, A. S., Chesnokov, M. S., Lazarevich, N. L.,
Penin, A. A., &amp; Logacheva, M. (2017). Effect of method of
deduplication on estimation of differential gene expression using
RNA-seq. <em>PeerJ</em>, <em>5</em>, e3091.
</div>
<div id="ref-koch2018beginner" class="csl-entry" role="listitem">
Koch, C. M., Chiu, S. F., Akbarpour, M., Bharat, A., Ridge, K. M.,
Bartom, E. T., &amp; Winter, D. R. (2018). A beginner’s guide to
analysis of RNA sequencing data. <em>American Journal of Respiratory
Cell and Molecular Biology</em>, <em>59</em>(2), 145–157.
</div>
<div id="ref-law2016rna" class="csl-entry" role="listitem">
Law, C. W., Alhamdoosh, M., Su, S., Smyth, G. K., &amp; Ritchie, M. E.
(2016). RNA-seq analysis is easy as 1-2-3 with limma, glimma and edgeR.
<em>F1000Research</em>, <em>5</em>.
</div>
<div id="ref-law2020guide" class="csl-entry" role="listitem">
Law, C. W., Zeglinski, K., Dong, X., Alhamdoosh, M., Smyth, G. K., &amp;
Ritchie, M. E. (2020). A guide to creating design matrices for gene
expression experiments. <em>F1000Research</em>, <em>9</em>.
</div>
<div id="ref-levin2010comprehensive" class="csl-entry" role="listitem">
Levin, J. Z., Yassour, M., Adiconis, X., Nusbaum, C., Thompson, D. A.,
Friedman, N., Gnirke, A., &amp; Regev, A. (2010). Comprehensive
comparative analysis of strand-specific RNA sequencing methods.
<em>Nature Methods</em>, <em>7</em>(9), 709–715.
</div>
<div id="ref-liu2016evaluation" class="csl-entry" role="listitem">
Liu, Q., &amp; Markatou, M. (2016). Evaluation of methods in removing
batch effects on RNA-seq data. <em>Infect Dis Transl Med</em>,
<em>2</em>(1), 3–9.
</div>
<div id="ref-liu2014rna" class="csl-entry" role="listitem">
Liu, Y., Zhou, J., &amp; White, K. P. (2014). RNA-seq differential
expression studies: More sequence or more replication?
<em>Bioinformatics</em>, <em>30</em>(3), 301–304.
</div>
<div id="ref-luecken2022benchmarking" class="csl-entry" role="listitem">
Luecken, M. D., Büttner, M., Chaichoompu, K., Danese, A., Interlandi,
M., Müller, M. F., Strobl, D. C., Zappia, L., Dugas, M., Colomé-Tatché,
M., et al. (2022). Benchmarking atlas-level data integration in
single-cell genomics. <em>Nature Methods</em>, <em>19</em>(1), 41–50.
</div>
<div id="ref-luecken2019current" class="csl-entry" role="listitem">
Luecken, M. D., &amp; Theis, F. J. (2019). Current best practices in
single-cell RNA-seq analysis: A tutorial. <em>Molecular Systems
Biology</em>, <em>15</em>(6), e8746.
</div>
<div id="ref-manimaran2016batchqc" class="csl-entry" role="listitem">
Manimaran, S., Selby, H. M., Okrah, K., Ruberman, C., Leek, J. T.,
Quackenbush, J., Haibe-Kains, B., Bravo, H. C., &amp; Johnson, W. E.
(2016). BatchQC: Interactive software for evaluating sample and batch
effects in genomic data. <em>Bioinformatics</em>, <em>32</em>(24),
3836–3838.
</div>
<div id="ref-marioni2008rna" class="csl-entry" role="listitem">
Marioni, J. C., Mason, C. E., Mane, S. M., Stephens, M., &amp; Gilad, Y.
(2008). RNA-seq: An assessment of technical reproducibility and
comparison with gene expression arrays. <em>Genome Research</em>,
<em>18</em>(9), 1509–1517.
</div>
<div id="ref-mehta2021asaim" class="csl-entry" role="listitem">
Mehta, S., Crane, M., Leith, E., Batut, B., Hiltemann, S., Arntzen, M.
Ø., Kunath, B. J., Pope, P. B., Delogu, F., Sajulga, R., et al. (2021).
ASaiM-MT: A validated and optimized ASaiM workflow for
metatranscriptomics analysis within galaxy framework.
<em>F1000Research</em>, <em>10</em>, 103.
</div>
<div id="ref-melsted2019modular" class="csl-entry" role="listitem">
Melsted, P., Booeshaghi, A. S., Gao, F., Beltrame, E., Lu, L.,
Hjorleifsson, K. E., Gehring, J., &amp; Pachter, L. (2019). Modular and
efficient pre-processing of single-cell RNA-seq. <em>BioRxiv</em>,
673285.
</div>
<div id="ref-moses2022museum" class="csl-entry" role="listitem">
Moses, L., &amp; Pachter, L. (2022). Museum of spatial transcriptomics.
<em>Nature Methods</em>, <em>19</em>(5), 534–546.
</div>
<div id="ref-nguyen2018experimental" class="csl-entry" role="listitem">
Nguyen, Q. H., Pervolarakis, N., Nee, K., &amp; Kessenbrock, K. (2018).
Experimental considerations for single-cell RNA sequencing approaches.
<em>Frontiers in Cell and Developmental Biology</em>, <em>6</em>, 108.
</div>
<div id="ref-ouyang2021shinycell" class="csl-entry" role="listitem">
Ouyang, J. F., Kamaraj, U. S., Cao, E. Y., &amp; Rackham, O. J. (2021).
ShinyCell: Simple and sharable visualization of single-cell gene
expression data. <em>Bioinformatics</em>, <em>37</em>(19), 3374–3376.
</div>
<div id="ref-parekh2016impact" class="csl-entry" role="listitem">
Parekh, S., Ziegenhain, C., Vieth, B., Enard, W., &amp; Hellmann, I.
(2016). The impact of amplification on differential expression analyses
by RNA-seq. <em>Scientific Reports</em>, <em>6</em>(1), 25533.
</div>
<div id="ref-schurch2016many" class="csl-entry" role="listitem">
Schurch, N. J., Schofield, P., Gierliński, M., Cole, C., Sherstnev, A.,
Singh, V., Wrobel, N., Gharbi, K., Simpson, G. G., Owen-Hughes, T., et
al. (2016). How many biological replicates are needed in an RNA-seq
experiment and which differential expression tool should you use?
<em>Rna</em>, <em>22</em>(6), 839–851.
</div>
<div id="ref-seyednasrollah2015comparison" class="csl-entry" role="listitem">
Seyednasrollah, F., Laiho, A., &amp; Elo, L. L. (2015). Comparison of
software packages for detecting differential expression in RNA-seq
studies. <em>Briefings in Bioinformatics</em>, <em>16</em>(1), 59–70.
</div>
<div id="ref-soneson2020exploremodelmatrix" class="csl-entry" role="listitem">
Soneson, C., Marini, F., Geier, F., Love, M. I., &amp; Stadler, M. B.
(2020). ExploreModelMatrix: Interactive exploration for improved
understanding of design matrices and linear models in r.
<em>F1000Research</em>, <em>9</em>(512), 512.
</div>
<div id="ref-sun2022comprehensive" class="csl-entry" role="listitem">
Sun, X., Lin, X., Li, Z., &amp; Wu, H. (2022). A comprehensive
comparison of supervised and unsupervised methods for cell type
identification in single-cell RNA-seq. <em>Briefings in
Bioinformatics</em>, <em>23</em>(2), bbab567.
</div>
<div id="ref-svensson2017power" class="csl-entry" role="listitem">
Svensson, V., Natarajan, K. N., Ly, L.-H., Miragaia, R. J., Labalette,
C., Macaulay, I. C., Cvejic, A., &amp; Teichmann, S. A. (2017). Power
analysis of single-cell RNA-sequencing experiments. <em>Nature
Methods</em>, <em>14</em>(4), 381–387.
</div>
<div id="ref-teng2016benchmark" class="csl-entry" role="listitem">
Teng, M., Love, M. I., Davis, C. A., Djebali, S., Dobin, A., Graveley,
B. R., Li, S., Mason, C. E., Olson, S., Pervouchine, D., et al. (2016).
A benchmark for RNA-seq quantification pipelines. <em>Genome
Biology</em>, <em>17</em>(1), 1–12.
</div>
<div id="ref-wagner2012measurement" class="csl-entry" role="listitem">
Wagner, G. P., Kin, K., &amp; Lynch, V. J. (2012). Measurement of mRNA
abundance using RNA-seq data: RPKM measure is inconsistent among
samples. <em>Theory in Biosciences</em>, <em>131</em>, 281–285.
</div>
<div id="ref-walker2022deciphering" class="csl-entry" role="listitem">
Walker, B. L., Cang, Z., Ren, H., Bourgain-Chang, E., &amp; Nie, Q.
(2022). Deciphering tissue structure and function using spatial
transcriptomics. <em>Communications Biology</em>, <em>5</em>(1), 220.
</div>
<div id="ref-wang2017comprehensive" class="csl-entry" role="listitem">
Wang, S., &amp; Gribskov, M. (2017). Comprehensive evaluation of de novo
transcriptome assembly programs and their effects on differential gene
expression analysis. <em>Bioinformatics</em>, <em>33</em>(3), 327–333.
</div>
<div id="ref-wang2019comparative" class="csl-entry" role="listitem">
Wang, T., Li, B., Nelson, C. E., &amp; Nabavi, S. (2019). Comparative
analysis of differential gene expression analysis tools for single-cell
RNA sequencing data. <em>BMC Bioinformatics</em>, <em>20</em>(1), 1–16.
</div>
<div id="ref-xi2021benchmarking" class="csl-entry" role="listitem">
Xi, N. M., &amp; Li, J. J. (2021). Benchmarking computational
doublet-detection methods for single-cell RNA sequencing data. <em>Cell
Systems</em>, <em>12</em>(2), 176–194.
</div>
<div id="ref-xie2021automatic" class="csl-entry" role="listitem">
Xie, B., Jiang, Q., Mora, A., &amp; Li, X. (2021). Automatic cell type
identification methods for single-cell RNA sequencing. <em>Computational
and Structural Biotechnology Journal</em>, <em>19</em>, 5874–5887.
</div>
<div id="ref-yue2023guidebook" class="csl-entry" role="listitem">
Yue, L., Liu, F., Hu, J., Yang, P., Wang, Y., Dong, J., Shu, W., Huang,
X., &amp; Wang, S. (2023). A guidebook of spatial transcriptomic
technologies, data resources and analysis approaches. <em>Computational
and Structural Biotechnology Journal</em>.
</div>
<div id="ref-zhang2020determining" class="csl-entry" role="listitem">
Zhang, M. J., Ntranos, V., &amp; Tse, D. (2020). Determining sequencing
depth in a single-cell RNA-seq experiment. <em>Nature
Communications</em>, <em>11</em>(1), 774.
</div>
<div id="ref-zhao2018rnaseqsamplesize" class="csl-entry" role="listitem">
Zhao, S., Li, C.-I., Guo, Y., Sheng, Q., &amp; Shyr, Y. (2018).
RnaSeqSampleSize: Real data based sample size estimation for RNA
sequencing. <em>BMC Bioinformatics</em>, <em>19</em>(1), 1–8.
</div>
<div id="ref-zhao2020misuse" class="csl-entry" role="listitem">
Zhao, S., Ye, Z., &amp; Stanton, R. (2020). Misuse of RPKM or TPM
normalization when comparing across samples and sequencing protocols.
<em>Rna</em>, <em>26</em>(8), 903–909.
</div>
<div id="ref-zhao2015comparison" class="csl-entry" role="listitem">
Zhao, S., Zhang, Y., Gordon, W., Quan, J., Xi, H., Du, S., Schack, D.
von, &amp; Zhang, B. (2015). Comparison of stranded and non-stranded
RNA-seq transcriptome profiling and investigation of gene overlap.
<em>BMC Genomics</em>, <em>16</em>(1), 1–14.
</div>
</div>



</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="../chapters/meta/index.html" class="pagination-link">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Meta Transcriptomics</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
  </div>
</nav>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">2023 Roy Francis</div>   
    <div class="nav-footer-center">
      &nbsp;
    <div class="toc-actions"><div><i class="bi bi-github"></i></div><div class="action-links"><p><a href="https://github.com/royfrancis/rskb/edit/main/chapters/references.qmd" class="toc-action">Edit this page</a></p><p><a href="https://github.com/royfrancis/rskb/issues/new" class="toc-action">Report an issue</a></p></div></div></div>
    <div class="nav-footer-right">Published with <a href="https://quarto.org/">Quarto</a> v1.3.340</div>
  </div>
</footer>



<script src="../site_libs/quarto-html/zenscroll-min.js"></script>
</body></html>