---
title: Spatial
description: Gene expression with spatial information
---

## Overview

There are three major resolution scales in ST data:

- multi-cell
- single-cell
- sub-cellular

@walker2022deciphering, @yue2023guidebook, @moses2022museum

## Techniques
### 10X Visium

- multi-cell resolution
- [Q&A](https://kb.10xgenomics.com/hc/en-us/categories/360002304851-Visium)
- [Spatial resolution and capture area](https://kb.10xgenomics.com/hc/en-us/articles/360035487572-What-is-the-spatial-resolution-and-configuration-of-the-capture-area-of-the-Visium-Gene-Expression-Slide-)
- Capture area: 6.5mm x 6.5mm
- Total spots: 4992
- Spot diameter: 55um

## Methods/Tools

### Frameworks

- Seurat
- Giotto
- Squidpy
- stLearn
- [Semla](https://ludvigla.github.io/semla/)

### SVG

Identification of spatially variable genes.

- SpatialDE
- SPARK
- SOMDE
- Sepal
- scGCO
- SpaGCN
- SpatialLIBD
- stLearn

### Spatial deconvolution

- STdeconvolve
  - Reference-free
  - https://github.com/JEFworks-Lab/STdeconvolve

- SpaceXR
  - Needs reference
  - Runs into error
  - https://github.com/dmcable/spacexr

- SPOTlight
  - Uses reference
  - https://github.com/MarcElosua/SPOTlight

- SpatialDWLS
- RCTD
- DSTG

### Cell interaction

Given the relative stability of cellular locations, spatial
transcriptomics allows us to reveal cellâ€“cell interactions (CCI),
also referred to as cell-cell communications (CCC), with fewer
false positives than similar analysis with scRNA-seq data.

- SpaOTsc

## Datasets

- R package [TENxVisiumData](https://bioconductor.org/packages/release/data/experiment/vignettes/TENxVisiumData/inst/doc/vignette.html)

## Resources

- [OSTA Book](https://lmweber.org/OSTA-book/)

## References {.unnumbered}
