<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.549">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="description" content="Single-cell RNA-Seq experiments and analyses">

<title>RSKB - 2&nbsp; Gene expression</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<link href="../../chapters/spatial/index.html" rel="next">
<link href="../../chapters/bulk/index.html" rel="prev">
<link href="../../assets/favicon.png" rel="icon" type="image/png">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<link href="../../site_libs/quarto-contrib/fontawesome6-0.1.0/all.css" rel="stylesheet">
<link href="../../site_libs/quarto-contrib/fontawesome6-0.1.0/latex-fontsize.css" rel="stylesheet">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<style>html{ scroll-behavior: smooth; }</style>
<link rel="preconnect" href="https://fonts.googleapis.com">
<link rel="preconnect" href="https://fonts.gstatic.com" crossorigin="">
<link href="https://fonts.googleapis.com/css2?family=JetBrains+Mono&amp;family=Nunito:ital,wght@0,400;0,600;1,400;1,600&amp;display=swap" rel="stylesheet">


</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top quarto-banner">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../chapters/sc/index.html"><strong>Single Cell RNASeq</strong></a></li><li class="breadcrumb-item"><a href="../../chapters/sc/index.html"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Gene expression</span></a></li></ol></nav>
        <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default page-columns page-full"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../chapters/sc/index.html"><strong>Single Cell RNASeq</strong></a></li><li class="breadcrumb-item"><a href="../../chapters/sc/index.html"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Gene expression</span></a></li></ol></nav>
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Gene expression</span></h1>
                  <div>
        <div class="description">
          Single-cell RNA-Seq experiments and analyses
        </div>
      </div>
                </div>
  </div>
    
  
  <div class="quarto-title-meta">

      
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../../">RSKB</a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/royfrancis/rskb" title="Source Code" class="quarto-navigation-tool px-1" aria-label="Source Code"><i class="bi bi-github"></i></a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Home</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
 <span class="menu-text"><strong>Bulk RNASeq</strong></span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../chapters/bulk/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Bulk gene expression</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">
 <span class="menu-text"><strong>Single Cell RNASeq</strong></span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../chapters/sc/index.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Gene expression</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../chapters/spatial/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Spatial</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../chapters/airr/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">AIRR-Seq</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../chapters/atacseq/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">ATAC-Seq</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true">
 <span class="menu-text"><strong>Meta Transcriptomics</strong></span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../chapters/meta/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Meta transcriptomics</span></span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#sc-overview" id="toc-sc-overview" class="nav-link active" data-scroll-target="#sc-overview"><span class="header-section-number">2.1</span> Overview</a></li>
  <li><a href="#sc-exp-design" id="toc-sc-exp-design" class="nav-link" data-scroll-target="#sc-exp-design"><span class="header-section-number">2.2</span> Experimental design</a>
  <ul class="collapse">
  <li><a href="#batch-design-number-of-cells" id="toc-batch-design-number-of-cells" class="nav-link" data-scroll-target="#batch-design-number-of-cells"><span class="header-section-number">2.2.1</span> Batch design, number of cells</a></li>
  <li><a href="#sequencing-depth" id="toc-sequencing-depth" class="nav-link" data-scroll-target="#sequencing-depth"><span class="header-section-number">2.2.2</span> Sequencing depth</a></li>
  </ul></li>
  <li><a href="#mapping-and-quantification" id="toc-mapping-and-quantification" class="nav-link" data-scroll-target="#mapping-and-quantification"><span class="header-section-number">2.3</span> Mapping and Quantification</a>
  <ul class="collapse">
  <li><a href="#cellranger" id="toc-cellranger" class="nav-link" data-scroll-target="#cellranger"><span class="header-section-number">2.3.1</span> CellRanger</a></li>
  <li><a href="#kallisto-bustools" id="toc-kallisto-bustools" class="nav-link" data-scroll-target="#kallisto-bustools"><span class="header-section-number">2.3.2</span> Kallisto Bustools</a></li>
  <li><a href="#salmon-alevin" id="toc-salmon-alevin" class="nav-link" data-scroll-target="#salmon-alevin"><span class="header-section-number">2.3.3</span> Salmon Alevin</a></li>
  <li><a href="#nextflow-nf-core-rnaseq" id="toc-nextflow-nf-core-rnaseq" class="nav-link" data-scroll-target="#nextflow-nf-core-rnaseq"><span class="header-section-number">2.3.4</span> Nextflow nf-core rnaseq</a></li>
  </ul></li>
  <li><a href="#doublet-detection" id="toc-doublet-detection" class="nav-link" data-scroll-target="#doublet-detection"><span class="header-section-number">2.4</span> Doublet detection</a></li>
  <li><a href="#cell-type-identification" id="toc-cell-type-identification" class="nav-link" data-scroll-target="#cell-type-identification"><span class="header-section-number">2.5</span> Cell type identification</a></li>
  <li><a href="#differential-expression" id="toc-differential-expression" class="nav-link" data-scroll-target="#differential-expression"><span class="header-section-number">2.6</span> Differential expression</a></li>
  <li><a href="#data-integration" id="toc-data-integration" class="nav-link" data-scroll-target="#data-integration"><span class="header-section-number">2.7</span> Data Integration</a></li>
  <li><a href="#trajectory" id="toc-trajectory" class="nav-link" data-scroll-target="#trajectory"><span class="header-section-number">2.8</span> Trajectory</a></li>
  <li><a href="#rna-velocity" id="toc-rna-velocity" class="nav-link" data-scroll-target="#rna-velocity"><span class="header-section-number">2.9</span> RNA velocity</a></li>
  <li><a href="#databases" id="toc-databases" class="nav-link" data-scroll-target="#databases"><span class="header-section-number">2.10</span> Databases</a>
  <ul class="collapse">
  <li><a href="#data" id="toc-data" class="nav-link" data-scroll-target="#data"><span class="header-section-number">2.10.1</span> Data</a></li>
  <li><a href="#markers" id="toc-markers" class="nav-link" data-scroll-target="#markers"><span class="header-section-number">2.10.2</span> Markers</a></li>
  </ul></li>
  <li><a href="#interactive-visualisation-frameworks" id="toc-interactive-visualisation-frameworks" class="nav-link" data-scroll-target="#interactive-visualisation-frameworks"><span class="header-section-number">2.11</span> Interactive visualisation frameworks</a></li>
  <li><a href="#learning" id="toc-learning" class="nav-link" data-scroll-target="#learning"><span class="header-section-number">2.12</span> Learning</a></li>
  <li><a href="#references" id="toc-references" class="nav-link" data-scroll-target="#references">References</a></li>
  </ul>
<div class="toc-actions"><ul><li><a href="https://github.com/royfrancis/rskb/edit/main/chapters/sc/index.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/royfrancis/rskb/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content">





<section id="sc-overview" class="level2" data-number="2.1">
<h2 data-number="2.1" class="anchored" data-anchor-id="sc-overview"><span class="header-section-number">2.1</span> Overview</h2>
<p><strong>Current best practices in scRNA-Seq</strong></p>
<ul>
<li>Perform QC by finding outlier peaks in the number of genes, the count depth and the fraction of mitochondrial reads. Consider these covariates jointly instead of separately.</li>
<li>Be as permissive of QC thresholding as possible, and revisit QC if downstream clustering cannot be interpreted.</li>
<li>If the distribution of QC covariates differ between samples, QC thresholds should be determined separately for each sample to account for sample quality differences</li>
<li>We recommend scran for normalization of non-full-length datasets. An alternative is to evaluate normalization approaches via scone especially for plate-based datasets. Full-length scRNA-seq protocols can be corrected for gene length using bulk methods.</li>
<li>There is no consensus on scaling genes to 0 mean and unit variance. We prefer not to scale gene expression.</li>
<li>Normalized data should be log(x+1)-transformed for use with downstream analysis methods that assume data are normally distributed.</li>
<li>Regress out biological covariates only for trajectory inference and if other biological processes of interest are not masked by the regressed out biological covariate.</li>
<li>Regress out technical and biological covariates jointly rather than serially.</li>
<li>Plate-based dataset pre-processing may require regressing out counts, normalization via non-linear normalization methods or downsampling.</li>
<li>We recommend performing batch correction via ComBat when cell type and state compositions between batches are consistent</li>
<li>Data integration and batch correction should be performed by different methods. Data integration tools may over-correct simple batch effects.</li>
<li>Users should be cautious of signals found only after expression recovery. Exploratory analysis may be best performed without this step.</li>
<li>We recommend selecting between 1,000 and 5,000 highly variable genes depending on dataset complexity.</li>
<li>Feature selection methods that use gene expression means and variances cannot be used when gene expression values have been normalized to zero mean and unit variance, or when residuals from model fitting are used as normalized expression values. Thus, one must consider what pre-processing to perform before selecting HVGs.</li>
<li>Dimensionality reduction methods should be considered separately for summarization and visualization.</li>
<li>We recommend UMAP for exploratory visualization; PCA for general purpose summarization; and diffusion maps as an alternative to PCA for trajectory inference summarization.</li>
<li>PAGA with UMAP is a suitable alternative to visualize particularly complex datasets.</li>
<li>Use measured data for statistical testing, corrected data for visual comparison of data and reduced data for other downstream analysis based on finding the underlying biological data manifold.</li>
<li>We recommend clustering by Louvain community detection on a single-cell KNN graph.</li>
<li>Clustering does not have to be performed at a single resolution. Subclustering particular cell clusters is a valid approach to focus on more detailed substructures in a dataset.</li>
<li>Do not use marker gene P-values to validate a cell-identity cluster, especially when the detected marker genes do not help to annotate the community. P-values may be inflated.</li>
<li>Note that marker genes for the same cell-identity cluster may differ between datasets purely due to dataset cell type and state compositions.</li>
<li>If relevant reference atlases exist, we recommend using automated cluster annotation combined with data-derived marker-gene-based manual annotation to annotate clusters.</li>
<li>Consider that statistical tests over changes in the proportion of a cell-identity cluster between samples are dependent on one another.</li>
<li>Inferred trajectories do not have to represent a biological process. Further sources of evidence should be collected to interpret a trajectory.</li>
<li>DE testing should not be performed on corrected data (denoised, batch corrected, etc.), but instead on measured data with technical covariates included in the model.</li>
<li>Users should not rely on DE testing tools to correct models with confounded covariates. Model specification should be performed carefully ensuring a full-rank design matrix.</li>
<li>We recommend using MAST or limma for DE testing.</li>
<li>Users should be wary of uncertainty in the inferred regulatory relationships. Modules of genes that are enriched for regulatory relationships will be more reliable than individual edges.</li>
</ul>
<p><span class="citation" data-cites="luecken2019current">Luecken &amp; Theis (<a href="#ref-luecken2019current" role="doc-biblioref">2019</a>)</span></p>
<p><strong>Best practices for single-cell analysis across modalities</strong></p>
<p><span class="citation" data-cites="heumos2023best">Heumos et al. (<a href="#ref-heumos2023best" role="doc-biblioref">2023</a>)</span></p>
<p><strong>What information should be included in an scRNA-Seq publication?</strong></p>
<p><span class="citation" data-cites="fullgrabe2020guidelines">Füllgrabe et al. (<a href="#ref-fullgrabe2020guidelines" role="doc-biblioref">2020</a>)</span></p>
</section>
<section id="sc-exp-design" class="level2" data-number="2.2">
<h2 data-number="2.2" class="anchored" data-anchor-id="sc-exp-design"><span class="header-section-number">2.2</span> Experimental design</h2>
<p><strong>Experimental Considerations for Single-Cell RNA Sequencing Approaches</strong></p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="https://www.frontiersin.org/files/Articles/391125/fcell-06-00108-HTML/image_m/fcell-06-00108-g001.jpg" class="img-fluid figure-img"></p>
<figcaption>Overview of step-wise approach to designing single-cell analysis workflows. RNA integrity number (RIN); Reads per cell (RPC).</figcaption>
</figure>
</div>
<p><span class="citation" data-cites="nguyen2018experimental">Nguyen et al. (<a href="#ref-nguyen2018experimental" role="doc-biblioref">2018</a>)</span></p>
<p><strong>How many reads are needed per cell? Sequencing depth?</strong></p>
<blockquote class="blockquote">
<p>Given a fixed budget, sequencing as many cells as possible at approximately one read per cell per gene is optimal, both theoretically and experimentally.</p>
</blockquote>
<p><span class="citation" data-cites="zhang2020determining">Zhang et al. (<a href="#ref-zhang2020determining" role="doc-biblioref">2020</a>)</span></p>
<section id="batch-design-number-of-cells" class="level3" data-number="2.2.1">
<h3 data-number="2.2.1" class="anchored" data-anchor-id="batch-design-number-of-cells"><span class="header-section-number">2.2.1</span> Batch design, number of cells</h3>
<p><strong>Avoid technical biases.</strong></p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="assets/baran-2018-1.jpg" class="img-fluid figure-img"></p>
<figcaption>Experimental design examples. In the confounded design, cells are isolated from each sample onto separate plates, processed at potentially different times and the two groups (indicated by different colors) are sequenced on separate lanes of the sequencer. In the balanced design on the right, all samples are evenly distributed across all stages of the experiment, thus reducing the sources of technical variation in the experiment.</figcaption>
</figure>
</div>
<p><strong>Deciding appropriate cell numbers</strong></p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="assets/baran-2018-2.jpg" class="img-fluid figure-img"></p>
<figcaption>Estimate of cells required for experiments with various parameters. (A) The plot shows the log10(#Cells) required to capture at least 50 cell types based on the parameters on the X- and Y-axes. (B) The plot shows the log10(#Cells) required to capture the number of cells on the Y-axis if the population consists of 20 cell types.</figcaption>
</figure>
</div>
<p><span class="citation" data-cites="baran2018experimental">Baran-Gale et al. (<a href="#ref-baran2018experimental" role="doc-biblioref">2018</a>)</span></p>
<ul>
<li><a href="https://satijalab.org/howmanycells/"><i class="fa-solid fa-toolbox" aria-label="toolbox"></i> SatijaLab Cells Calculator</a></li>
<li><a href="https://github.com/bvieth/powsimR"><i class="fa-solid fa-toolbox" aria-label="toolbox"></i> powsimR (R)</a></li>
</ul>
</section>
<section id="sequencing-depth" class="level3" data-number="2.2.2">
<h3 data-number="2.2.2" class="anchored" data-anchor-id="sequencing-depth"><span class="header-section-number">2.2.2</span> Sequencing depth</h3>
<blockquote class="blockquote">
<p>While 250 000 reads per cell are sufficient for accuracy, 1 million reads per cell were a good target for saturated gene detection.</p>
</blockquote>
<p><span class="citation" data-cites="svensson2017power">Svensson et al. (<a href="#ref-svensson2017power" role="doc-biblioref">2017</a>)</span></p>
<ul>
<li><a href="https://10xqc.com/"><i class="fa-solid fa-toolbox" aria-label="toolbox"></i> Compare 10X QC reports</a></li>
<li>SatijaLab <a href="https://satijalab.org/costpercell/"><i class="fa-solid fa-toolbox" aria-label="toolbox"></i> Cost Per Cell Calculator</a></li>
</ul>
</section>
</section>
<section id="mapping-and-quantification" class="level2" data-number="2.3">
<h2 data-number="2.3" class="anchored" data-anchor-id="mapping-and-quantification"><span class="header-section-number">2.3</span> Mapping and Quantification</h2>
<section id="cellranger" class="level3" data-number="2.3.1">
<h3 data-number="2.3.1" class="anchored" data-anchor-id="cellranger"><span class="header-section-number">2.3.1</span> CellRanger</h3>
<ul>
<li><p>Process chromium data</p></li>
<li><p>BCL to FASTQ</p></li>
<li><p>FASTQ to cellxgene counts</p></li>
<li><p>Feature barcoding</p></li>
<li><p><a href="https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/what-is-cell-ranger"><i class="fa-solid fa-toolbox" aria-label="toolbox"></i> CellRanger</a></p></li>
</ul>
</section>
<section id="kallisto-bustools" class="level3" data-number="2.3.2">
<h3 data-number="2.3.2" class="anchored" data-anchor-id="kallisto-bustools"><span class="header-section-number">2.3.2</span> Kallisto Bustools</h3>
<ul>
<li><p>10x, inDrop and Dropseq</p></li>
<li><p>Generate cellxgene, cellxtranscript matrix</p></li>
<li><p>RNA velocity data</p></li>
<li><p>Feature barcoding</p></li>
<li><p>QC reports</p></li>
<li><p><a href="https://bustools.github.io/"><i class="fa-solid fa-toolbox" aria-label="toolbox"></i> BUSTools</a></p></li>
</ul>
<p><span class="citation" data-cites="melsted2019modular">Melsted et al. (<a href="#ref-melsted2019modular" role="doc-biblioref">2019</a>)</span></p>
</section>
<section id="salmon-alevin" class="level3" data-number="2.3.3">
<h3 data-number="2.3.3" class="anchored" data-anchor-id="salmon-alevin"><span class="header-section-number">2.3.3</span> Salmon Alevin</h3>
<ul>
<li>Drop-seq, 10x-Chromium v1/2/3, inDropV2, CELSeq 1/2, Quartz-Seq2, sci-RNA-seq3</li>
<li>Generate cellxgene matrix</li>
<li><a href="https://salmon.readthedocs.io/en/latest/alevin.html"><i class="fa-solid fa-toolbox" aria-label="toolbox"></i> Alevin</a></li>
</ul>
</section>
<section id="nextflow-nf-core-rnaseq" class="level3" data-number="2.3.4">
<h3 data-number="2.3.4" class="anchored" data-anchor-id="nextflow-nf-core-rnaseq"><span class="header-section-number">2.3.4</span> Nextflow nf-core rnaseq</h3>
<ul>
<li>Bulk RNA-Seq, SMART-Seq</li>
<li>QC, trimming, UMI demultiplexing, mapping, quantification</li>
<li>cellxgene matrix</li>
<li><a href="https://nf-co.re/scrnaseq"><i class="fa-solid fa-toolbox" aria-label="toolbox"></i> nf-core scrnaseq</a></li>
</ul>
</section>
</section>
<section id="doublet-detection" class="level2" data-number="2.4">
<h2 data-number="2.4" class="anchored" data-anchor-id="doublet-detection"><span class="header-section-number">2.4</span> Doublet detection</h2>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="https://ars.els-cdn.com/content/image/1-s2.0-S2405471220304592-gr5.jpg" class="img-fluid figure-img"></p>
<figcaption>Summary of doublet detection tools.</figcaption>
</figure>
</div>
<p><span class="citation" data-cites="xi2021benchmarking">Xi &amp; Li (<a href="#ref-xi2021benchmarking" role="doc-biblioref">2021</a>)</span></p>
</section>
<section id="cell-type-identification" class="level2" data-number="2.5">
<h2 data-number="2.5" class="anchored" data-anchor-id="cell-type-identification"><span class="header-section-number">2.5</span> Cell type identification</h2>
<p>Identification of cell types can be completely automated (by comparing to reference data/databases) or semi-automated (reference data + marker genes).</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="https://ars.els-cdn.com/content/image/1-s2.0-S2001037021004499-gr5_lrg.jpg" class="img-fluid figure-img"></p>
<figcaption>Summary of performance of the automatic cell-type identification methods. Bar graphs of the automatic cell-type identification methods with six evaluation criteria indicated. For each evaluation criteria, the length of the bars shows the performance of the automatic method: poor, median or good. The automatic methods are sorted based on the mean performance of the evaluation criteria. No bar: not evaluated.</figcaption>
</figure>
</div>
<p><span class="citation" data-cites="xie2021automatic">Xie et al. (<a href="#ref-xie2021automatic" role="doc-biblioref">2021</a>)</span>, <span class="citation" data-cites="sun2022comprehensive">Sun et al. (<a href="#ref-sun2022comprehensive" role="doc-biblioref">2022</a>)</span></p>
<ul>
<li><a href="https://github.com/jcao89757/SCINA"><i class="fa-solid fa-toolbox" aria-label="toolbox"></i> SCINA</a></li>
<li><a href="https://bioconductor.org/packages/release/bioc/html/SingleR.html"><i class="fa-solid fa-toolbox" aria-label="toolbox"></i> SingleR</a></li>
</ul>
</section>
<section id="differential-expression" class="level2" data-number="2.6">
<h2 data-number="2.6" class="anchored" data-anchor-id="differential-expression"><span class="header-section-number">2.6</span> Differential expression</h2>
<ul>
<li>Comparison of DGE tools for single-cell data</li>
</ul>
<blockquote class="blockquote">
<p>All of the tools perform well when there is no multimodality or low levels of multimodality. They all also perform better when the sparsity (zero counts) is less. For data with a high level of multimodality, methods that consider the behavior of each individual gene, such as DESeq2, EMDomics, Monocle2, DEsingle, and SigEMD, show better TPRs. If the level of multimodality is low, however, SCDE, MAST, and edgeR can provide higher precision.</p>
</blockquote>
<blockquote class="blockquote">
<p>In general, the methods that can capture multimodality (non-parametric methods), perform better than do the model-based methods designed for handling zero counts. However, a model-based method that can model the drop-out events well, can perform better in terms of true positive and false positive. We observed that methods developed specifically for scRNAseq data do not show significantly better performance compared to the methods designed for bulk RNAseq data; and methods that consider behavior of each individual gene (not all genes) in calling DE genes outperform the other tools.</p>
</blockquote>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="https://media.springernature.com/full/springer-static/image/art%3A10.1186%2Fs12859-019-2599-6/MediaObjects/12859_2019_2599_Fig7_HTML.png" class="img-fluid figure-img"></p>
<figcaption>Effect of sample size (number of cells) on detecting DE genes. The sample size is in horizontal axis, from 10 to 400 cells in each condition. Effect of sample size on a TPR, b FPR, c accuracy (=(TP + TN)/(TP + FP + TN + FN)), and precision (=TP/(TP + FP)). A threshold of 0.05 is used for FDR or adjusted p-value.</figcaption>
</figure>
</div>
<p><span class="citation" data-cites="wang2019comparative">Wang et al. (<a href="#ref-wang2019comparative" role="doc-biblioref">2019</a>)</span></p>
<ul>
<li>Differential expression without clustering or grouping</li>
<li><a href="https://github.com/alexisvdb/singleCellHaystack"><i class="fa-solid fa-toolbox" aria-label="toolbox"></i> singleCellHaystack</a></li>
</ul>
</section>
<section id="data-integration" class="level2" data-number="2.7">
<h2 data-number="2.7" class="anchored" data-anchor-id="data-integration"><span class="header-section-number">2.7</span> Data Integration</h2>
<ul>
<li>Single-cell data integration challenges</li>
</ul>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="assets/argelaguet-2021-tab1.png" class="img-fluid figure-img"></p>
<figcaption>Overview of common data integration methods classified according to their anchor choice.</figcaption>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="assets/argelaguet-2021-1.png" class="img-fluid figure-img"></p>
<figcaption>a–c, Depending on the anchor choice, three types of data integration strategies can be considered: horizontal integration with features as the anchors (a), vertical integration with cells as the anchors (b) and diagonal integration with no anchors in high-dimensional space (c). The left column shows the data modalities extracted, while the right column illustrates the resulting data matrices to be integrated, depending on the anchor choice.</figcaption>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="assets/argelaguet-2021-2.png" class="img-fluid figure-img"></p>
<figcaption>Mosaic integration. a, Overview of an experimental design where different data modalities (each block in the rows) are profiled in different subsets of cells (each block in the columns). Transparent matrices denote missing information. b, Resulting data matrices after applying a mosaic integration approach aimed at imputing missing data modalities.</figcaption>
</figure>
</div>
<p><span class="citation" data-cites="argelaguet2021computational">Argelaguet et al. (<a href="#ref-argelaguet2021computational" role="doc-biblioref">2021</a>)</span></p>
<ul>
<li>Comparison of data integration methods</li>
</ul>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="https://media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fs41592-021-01336-8/MediaObjects/41592_2021_1336_Fig2_HTML.png?as=webp" class="img-fluid figure-img"></p>
<figcaption>a, Overview of top and bottom ranked methods by overall score for the human immune cell task. Metrics are divided into batch correction (blue) and bio-conservation (pink) categories. Overall scores are computed using a 40/60 weighted mean of these category scores (see Methods for further visualization details and Supplementary Fig. 2 for the full plot). b,c, Visualization of the four best performers on the human immune cell integration task colored by cell identity (b) and batch annotation (c). The plots show uniform manifold approximation and projection layouts for the unintegrated data (left) and the top four performers (right).</figcaption>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="https://media.springernature.com/full/springer-static/image/art%3A10.1038%2Fs41592-021-01336-8/MediaObjects/41592_2021_1336_Fig3_HTML.png?as=webp" class="img-fluid figure-img"></p>
<figcaption>a, Scatter plot of the mean overall batch correction score against mean overall bio-conservation score for the selected methods on RNA tasks. Error bars indicate the standard error across tasks on which the methods ran. b, The overall scores for the best performing method, preprocessing and output combinations on each task as well as their usability and scalability. Methods that failed to run for a particular task were assigned the unintegrated ranking for that task. An asterisk after the method name (scANVI and scGen) indicates that, in addition, cell identity information was passed to this method. For ComBat and MNN, usability and scalability scores corresponding to the Python implementation of the methods are reported (Scanpy and mnnpy, respectively).</figcaption>
</figure>
</div>
<p><span class="citation" data-cites="luecken2022benchmarking">Luecken et al. (<a href="#ref-luecken2022benchmarking" role="doc-biblioref">2022</a>)</span></p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="https://media.springernature.com/full/springer-static/image/art%3A10.1186%2Fs13059-019-1850-9/MediaObjects/13059_2019_1850_Fig18_HTML.png?as=webp" class="img-fluid figure-img"></p>
<figcaption>Qualitative evaluation of 14 batch-effect correction methods using UMAP visualization for dataset 9 of human cell atlas. The 14 methods are organized into two panels, with the top panel showing UMAP plots of raw data, Seurat 2, Seurat 3, Harmony, fastMNN, MNN Correct, ComBat, and limma outputs, while the bottom panel shows the UMAP plots of scGen, Scanorama, MMD-ResNet, ZINB-WaVE, scMerge, LIGER, and BBKNN outputs. Cells are colored by batch.</figcaption>
</figure>
</div>
<blockquote class="blockquote">
<p>We tested 14 state-of-the-art batch correction algorithms designed to handle single-cell transcriptomic data. We found that each batch-effect removal method has its advantages and limitations, with no clearly superior method. Based on our results, we found LIGER, Harmony, and Seurat 3 to be the top batch mixing methods. Harmony performed well on datasets with common cell types, and also different technologies. The comparatively low runtime of Harmony also makes it suitable for initial data exploration of large datasets. Likewise, LIGER performed well, especially on datasets with non-identical cell types. The main drawback of LIGER is its longer runtime than Harmony, though it is acceptable for its performance. Seurat 3 is also able to handle large datasets, however with 20–50% longer runtime than LIGER. Due to its good batch mixing results with multiple batches, it is also recommended for such scenarios. To improve recovery of DEGs in batch-corrected data, we recommend scMerge for batch correction.</p>
</blockquote>
<p><span class="citation" data-cites="tran2020benchmark">Tran et al. (<a href="#ref-tran2020benchmark" role="doc-biblioref">2020</a>)</span></p>
</section>
<section id="trajectory" class="level2" data-number="2.8">
<h2 data-number="2.8" class="anchored" data-anchor-id="trajectory"><span class="header-section-number">2.8</span> Trajectory</h2>
<div id="fig-sc-trajectory-comparison" class="quarto-layout-panel">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-sc-trajectory-comparison-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-layout-row">
<div class="quarto-layout-cell" style="flex-basis: 50.0%;justify-content: center;">
<p><img src="https://media.springernature.com/full/springer-static/image/art%3A10.1038%2Fs41587-019-0071-9/MediaObjects/41587_2019_71_Fig2_HTML.png?as=webp" class="img-fluid figure-img"></p>
</div>
<div class="quarto-layout-cell" style="flex-basis: 50.0%;justify-content: center;">
<p><img src="https://media.springernature.com/full/springer-static/image/art%3A10.1038%2Fs41587-019-0071-9/MediaObjects/41587_2019_71_Fig3_HTML.png?as=webp" class="img-fluid figure-img"></p>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-sc-trajectory-comparison-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<strong>Fig</strong>&nbsp;2.1<strong>: </strong> Comparison of Trajectory inference methods.
</figcaption>
</figure>
</div>
<p><span class="citation" data-cites="saelens2019comparison">Saelens et al. (<a href="#ref-saelens2019comparison" role="doc-biblioref">2019</a>)</span></p>
<ul>
<li><a href="https://github.com/BaderLab/Tempora"><i class="fa-solid fa-toolbox" aria-label="toolbox"></i> Tempora</a> Trajectory inference for time-series data</li>
</ul>
</section>
<section id="rna-velocity" class="level2" data-number="2.9">
<h2 data-number="2.9" class="anchored" data-anchor-id="rna-velocity"><span class="header-section-number">2.9</span> RNA velocity</h2>
</section>
<section id="databases" class="level2" data-number="2.10">
<h2 data-number="2.10" class="anchored" data-anchor-id="databases"><span class="header-section-number">2.10</span> Databases</h2>
<section id="data" class="level3" data-number="2.10.1">
<h3 data-number="2.10.1" class="anchored" data-anchor-id="data"><span class="header-section-number">2.10.1</span> Data</h3>
<p>Single-cell data repositiories.</p>
<ul>
<li><a href="https://bioinfo.uth.edu/scrnaseqdb/">scRNASeqDB</a></li>
<li><a href="https://singlecell.broadinstitute.org/single_cell">Broad SIngleCell portal</a></li>
<li><a href="https://hemberg-lab.github.io/scRNA.seq.datasets/">Hemberg Lab collection</a></li>
<li><a href="https://www.10xgenomics.com/resources/datasets">10x datasets</a></li>
<li><a href="https://www.ebi.ac.uk/gxa/sc/home">EBI Cell Atlas</a></li>
<li><a href="https://jhubiostatistics.shinyapps.io/recount/">recount 2</a></li>
<li><a href="https://jinglebells.bgu.ac.il/">JingleBells</a></li>
<li><a href="https://db.cngb.org/cdcp/explore">CNGB</a></li>
<li><a href="https://bioconductor.org/packages/release/data/experiment/vignettes/TENxVisiumData/inst/doc/vignette.html">R TENxVisiumData</a></li>
</ul>
</section>
<section id="markers" class="level3" data-number="2.10.2">
<h3 data-number="2.10.2" class="anchored" data-anchor-id="markers"><span class="header-section-number">2.10.2</span> Markers</h3>
<p>Curated list of marker genes by organism, tissue and cell type.</p>
<ul>
<li><a href="https://panglaodb.se/">PanglaoDB</a></li>
<li><a href="http://biocc.hrbmu.edu.cn/CellMarker/browse.jsp">CellMarker</a></li>
</ul>
</section>
</section>
<section id="interactive-visualisation-frameworks" class="level2" data-number="2.11">
<h2 data-number="2.11" class="anchored" data-anchor-id="interactive-visualisation-frameworks"><span class="header-section-number">2.11</span> Interactive visualisation frameworks</h2>
<ul>
<li><a href="https://github.com/SGDDNB/ShinyCell"><i class="fa-solid fa-toolbox" aria-label="toolbox"></i> ShinyCell</a>(R Shiny)</li>
<li><a href="https://github.com/romanhaa/Cerebro"><i class="fa-solid fa-toolbox" aria-label="toolbox"></i> Cerebro</a>(R Shiny)</li>
<li><a href="http://vitessce.io/"><i class="fa-solid fa-toolbox" aria-label="toolbox"></i> Vitessce</a>(R, Python)</li>
<li><a href="https://db.cngb.org/cdcp/"><i class="fa-solid fa-toolbox" aria-label="toolbox"></i> CDCP</a></li>
<li><a href="https://singlecellvr.pinellolab.partners.org/"><i class="fa-solid fa-toolbox" aria-label="toolbox"></i> SingleCellVR</a></li>
</ul>
<p><img src="https://github.com/SGDDNB/ShinyCell/raw/master/images/comparison.png" class="img-fluid"></p>
<p><span class="citation" data-cites="ouyang2021shinycell">Ouyang et al. (<a href="#ref-ouyang2021shinycell" role="doc-biblioref">2021</a>)</span></p>
<p><img src="https://www.ncbi.nlm.nih.gov/labs/pmc/articles/PMC7391988/bin/lqaa052tbl1.jpg" class="img-fluid"></p>
<p><span class="citation" data-cites="cakir2020comparison">Cakir et al. (<a href="#ref-cakir2020comparison" role="doc-biblioref">2020</a>)</span></p>
</section>
<section id="learning" class="level2" data-number="2.12">
<h2 data-number="2.12" class="anchored" data-anchor-id="learning"><span class="header-section-number">2.12</span> Learning</h2>
<ul>
<li><a href="https://hbctraining.github.io/scRNA-seq_online/schedule/">HBC Training</a></li>
<li><a href="https://nbisweden.github.io/workshop-scRNAseq">NBIS workshop</a></li>
<li><a href="https://satijalab.org/seurat/index.html">Seurat tutorials</a></li>
</ul>
</section>
<section id="references" class="level2 unnumbered">
<h2 class="unnumbered anchored" data-anchor-id="references">References</h2>


<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" data-line-spacing="2" role="list">
<div id="ref-argelaguet2021computational" class="csl-entry" role="listitem">
Argelaguet, R., Cuomo, A. S., Stegle, O., &amp; Marioni, J. C. (2021). Computational principles and challenges in single-cell data integration. <em>Nature Biotechnology</em>, <em>39</em>(10), 1202–1215.
</div>
<div id="ref-baran2018experimental" class="csl-entry" role="listitem">
Baran-Gale, J., Chandra, T., &amp; Kirschner, K. (2018). Experimental design for single-cell RNA sequencing. <em>Briefings in Functional Genomics</em>, <em>17</em>(4), 233–239.
</div>
<div id="ref-cakir2020comparison" class="csl-entry" role="listitem">
Cakir, B., Prete, M., Huang, N., Van Dongen, S., Pir, P., &amp; Kiselev, V. Y. (2020). Comparison of visualization tools for single-cell RNAseq data. <em>NAR Genomics and Bioinformatics</em>, <em>2</em>(3), lqaa052.
</div>
<div id="ref-fullgrabe2020guidelines" class="csl-entry" role="listitem">
Füllgrabe, A., George, N., Green, M., Nejad, P., Aronow, B., Fexova, S. K., Fischer, C., Freeberg, M. A., Huerta, L., Morrison, N., et al. (2020). Guidelines for reporting single-cell RNA-seq experiments. <em>Nature Biotechnology</em>, <em>38</em>(12), 1384–1386.
</div>
<div id="ref-heumos2023best" class="csl-entry" role="listitem">
Heumos, L., Schaar, A. C., Lance, C., Litinetskaya, A., Drost, F., Zappia, L., Lücken, M. D., Strobl, D. C., Henao, J., Curion, F., et al. (2023). Best practices for single-cell analysis across modalities. <em>Nature Reviews Genetics</em>, 1–23.
</div>
<div id="ref-luecken2022benchmarking" class="csl-entry" role="listitem">
Luecken, M. D., Büttner, M., Chaichoompu, K., Danese, A., Interlandi, M., Müller, M. F., Strobl, D. C., Zappia, L., Dugas, M., Colomé-Tatché, M., et al. (2022). Benchmarking atlas-level data integration in single-cell genomics. <em>Nature Methods</em>, <em>19</em>(1), 41–50.
</div>
<div id="ref-luecken2019current" class="csl-entry" role="listitem">
Luecken, M. D., &amp; Theis, F. J. (2019). Current best practices in single-cell RNA-seq analysis: A tutorial. <em>Molecular Systems Biology</em>, <em>15</em>(6), e8746.
</div>
<div id="ref-melsted2019modular" class="csl-entry" role="listitem">
Melsted, P., Booeshaghi, A. S., Gao, F., Beltrame, E., Lu, L., Hjorleifsson, K. E., Gehring, J., &amp; Pachter, L. (2019). Modular and efficient pre-processing of single-cell RNA-seq. <em>BioRxiv</em>, 673285.
</div>
<div id="ref-nguyen2018experimental" class="csl-entry" role="listitem">
Nguyen, Q. H., Pervolarakis, N., Nee, K., &amp; Kessenbrock, K. (2018). Experimental considerations for single-cell RNA sequencing approaches. <em>Frontiers in Cell and Developmental Biology</em>, <em>6</em>, 108.
</div>
<div id="ref-ouyang2021shinycell" class="csl-entry" role="listitem">
Ouyang, J. F., Kamaraj, U. S., Cao, E. Y., &amp; Rackham, O. J. (2021). ShinyCell: Simple and sharable visualization of single-cell gene expression data. <em>Bioinformatics</em>, <em>37</em>(19), 3374–3376.
</div>
<div id="ref-saelens2019comparison" class="csl-entry" role="listitem">
Saelens, W., Cannoodt, R., Todorov, H., &amp; Saeys, Y. (2019). A comparison of single-cell trajectory inference methods. <em>Nature Biotechnology</em>, <em>37</em>(5), 547–554.
</div>
<div id="ref-sun2022comprehensive" class="csl-entry" role="listitem">
Sun, X., Lin, X., Li, Z., &amp; Wu, H. (2022). A comprehensive comparison of supervised and unsupervised methods for cell type identification in single-cell RNA-seq. <em>Briefings in Bioinformatics</em>, <em>23</em>(2), bbab567.
</div>
<div id="ref-svensson2017power" class="csl-entry" role="listitem">
Svensson, V., Natarajan, K. N., Ly, L.-H., Miragaia, R. J., Labalette, C., Macaulay, I. C., Cvejic, A., &amp; Teichmann, S. A. (2017). Power analysis of single-cell RNA-sequencing experiments. <em>Nature Methods</em>, <em>14</em>(4), 381–387.
</div>
<div id="ref-tran2020benchmark" class="csl-entry" role="listitem">
Tran, H. T. N., Ang, K. S., Chevrier, M., Zhang, X., Lee, N. Y. S., Goh, M., &amp; Chen, J. (2020). A benchmark of batch-effect correction methods for single-cell RNA sequencing data. <em>Genome Biology</em>, <em>21</em>, 1–32.
</div>
<div id="ref-wang2019comparative" class="csl-entry" role="listitem">
Wang, T., Li, B., Nelson, C. E., &amp; Nabavi, S. (2019). Comparative analysis of differential gene expression analysis tools for single-cell RNA sequencing data. <em>BMC Bioinformatics</em>, <em>20</em>(1), 1–16.
</div>
<div id="ref-xi2021benchmarking" class="csl-entry" role="listitem">
Xi, N. M., &amp; Li, J. J. (2021). Benchmarking computational doublet-detection methods for single-cell RNA sequencing data. <em>Cell Systems</em>, <em>12</em>(2), 176–194.
</div>
<div id="ref-xie2021automatic" class="csl-entry" role="listitem">
Xie, B., Jiang, Q., Mora, A., &amp; Li, X. (2021). Automatic cell type identification methods for single-cell RNA sequencing. <em>Computational and Structural Biotechnology Journal</em>, <em>19</em>, 5874–5887.
</div>
<div id="ref-zhang2020determining" class="csl-entry" role="listitem">
Zhang, M. J., Ntranos, V., &amp; Tse, D. (2020). Determining sequencing depth in a single-cell RNA-seq experiment. <em>Nature Communications</em>, <em>11</em>(1), 774.
</div>
</div>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="../../chapters/bulk/index.html" class="pagination-link  aria-label=" &lt;span="" gene="" expression&lt;="" span&gt;"="">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Bulk gene expression</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../../chapters/spatial/index.html" class="pagination-link" aria-label="<span class='chapter-number'>3</span>&nbsp; <span class='chapter-title'>Spatial</span>">
        <span class="nav-page-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Spatial</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
<p>2024 Roy Francis</p>
</div>   
    <div class="nav-footer-center">
      &nbsp;
    <div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/royfrancis/rskb/edit/main/chapters/sc/index.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/royfrancis/rskb/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></div>
    <div class="nav-footer-right">
<p>Published with <a href="https://quarto.org/">Quarto</a> v1.4.549
</p>
</div>
  </div>
</footer>




<script src="../../site_libs/quarto-html/zenscroll-min.js"></script>
</body></html>